plugins/tools/plugin-cache/src/node/utils.ts

```diff
@@ -0,0 +1,19 @@
+import { createHash } from 'node:crypto'
+import { fs } from 'vuepress/utils'
+
+export const hash = (data: string): string =>
+  createHash('md5').update(data).digest('hex')
+
+export const readFile = async <T = any>(filepath: string): Promise<T> => {
+  try {
+    const content = await fs.readFile(filepath, 'utf-8')
+    return JSON.parse(content) as T
+  } catch {
+    return {} as T
```

Comment:
I would prefer null here for this api,.the needed of empty object might be better placed outside this function.